include/z64ocarina.h

```diff
@@ -32,10 +32,135 @@ typedef enum {
     /* 24 */ OCARINA_SONG_SCARECROW_LONG = OCARINA_SONG_MAX // anything larger than 24 is considered the long scarecrow's song
 } OcarinaSongId;
 
-typedef enum OcarinaMode {
-    /* 0x18 */ OCARINA_MODE_APPLY_INV_SOT_FAST = 24,
+typedef enum {
+    /* 0x00 */ OCARINA_ACTION_0, // acts like free play but never set
+    /* 0x01 */ OCARINA_ACTION_FREE_PLAY,
+    /* 0x02 */ OCARINA_ACTION_DEMONSTRATE_SONATA, // Song demonstrations by teachers
```

Comment:
I think the demonstrate/prompt/check stuff deserves a longer comment maybe above the enum. I think I understand what "demonstrate" means (based on `func_80152CAC`, it seems that it's used for when various characters play a song for you. But I'm having trouble determining what a "check" value really means, and for "prompt", I'm not sure when you end up in that state. Is the ocarina action set to `OCARINA_ACTION_PROMPT_SONATA` when the staff appears and tells you when to play it? When the player successfully inputs it? When the notes start playing back?

Maybe this is really understandable to others already, or maybe we just don't yet know the answers to some of these questions, in which case I'm fine just leaving this as-is. But I feel this could really benefit from giving a little more thorough explanation of the various "categories".

Reply:
Relevant discussion with explanation of each term:
https://discord.com/channels/688807550715560050/915273929369731173/1062382682778112030

I agree the names need looking over, and in fact these names are out of sync with the audio ocarina code as well. Would definitely support adding comments, I'll see what I can come up with

Reply:
Any suggestions engineer124?